src/api/XRSession.js

```diff
@@ -259,59 +354,30 @@ export default class XRSession extends EventTarget {
       return;
     }
 
-    // If the session is suspended and we have a previously saved
-    // suspendedCallback, abort this call
-    if (this[PRIVATE].suspended && this[PRIVATE].suspendedCallback) {
-      return;
-    }
-
-    // Otherwise, if the session is suspended but has not yet creating
-    // the suspended callback, do so. It will resume once it is no
-    // longer suspended.
-    if (this[PRIVATE].suspended && !this[PRIVATE].suspendedCallback) {
-      this[PRIVATE].suspendedCallback = callback;
+    // Add callback to the queue and return its handle
+    const handle = ++this[PRIVATE].frameHandle;
+    const callbacks = this[PRIVATE].frameCallbacks;
+    if (callbacks.findIndex(d => d.callback === callback) === -1) {
```

Comment:
Ugh that was boneheaded, thanks for catching it. Too many times implementing addEventListener I guess. ;) Among other issues this would have resulted in a handle that could not be cancelled.